nova/virt/vmwareapi/vmops.py

```diff
@@ -2058,13 +2137,87 @@ def migrate_disk_and_power_off(self, context, instance, dest, flavor):
                                        total_steps=RESIZE_TOTAL_STEPS)
 
         # 1. Power off the instance
-        vm_util.power_off_instance(self._session, instance, vm_ref)
         self._update_instance_progress(context, instance,
                                        step=1,
                                        total_steps=RESIZE_TOTAL_STEPS)
+        vm_util.power_off_instance(self._session, instance, vm_ref)
+
+        # 2. Detach the volumes
+        self._update_instance_progress(context, instance,
+                                       step=2,
+                                       total_steps=RESIZE_TOTAL_STEPS)
+        self._detach_volumes(instance, block_device_info)
```

Comment:
Can we detect if detach failed for a volume and attache the other ones?

Reply:
Is this just for keeping the VM in a "better than nothing" state ?